### PR TITLE
Car enemy + movement compoent

### PR DESCRIPTION
**Pull request by Tomerd-afk**

### CODE REVIEW

File: Assets/Resources/AiClasses/CarAi.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Comment (toryjleo) [inline Assets/Resources/AiClasses/CarAi.cs:4]: Ditto

File: Assets/Resources/AiClasses/BikeAI.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Comment (toryjleo) [inline Assets/Resources/AiClasses/BikeAI.cs:4]: Please leave class comment

File: Assets/Resources/AiClasses/CarAi.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    // Start is called before the first frame update
+    void Awake()
+    {
+        movementComponent = GetComponent<CarMovementComponent>();
+    }
+
+    // Update is called once per frame
```
Comment (toryjleo) [inline Assets/Resources/AiClasses/CarAi.cs:31]: Please leave method comment

Comment (toryjleo) [reply Assets/Resources/AiClasses/CarAi.cs:31]: This is not "Update"

File: Assets/Scripts/AI/MovementComponent.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class MovementComponent : MonoBehaviour
```
Comment (toryjleo) [inline Assets/Scripts/AI/MovementComponent.cs:5]: Ditto

File: Assets/Scripts/AI/MovementComponent.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    #endregion
+
+    public virtual void FixedUpdate()
+    {
+        verticalInput = Input.GetAxis("Vertical");
+        horizontalInput = Input.GetAxis("Horizontal");
+        ApplyForces();
+    }
+
+    public virtual void ApplyForces()
```
Comment (toryjleo) [inline Assets/Scripts/AI/MovementComponent.cs:90]: Please leave method comment

File: Assets/TrackingScript.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[System.Serializable]
+public class TrackingScript : MonoBehaviour
```
Comment (toryjleo) [inline Assets/TrackingScript.cs:6]: Ditto

**Review by toryjleo — CHANGES_REQUESTED** [review]
There are further issues when I play this branch:

- The old non-vehicle enemies do not spawn in
- The car does not appear to move correctly, it rotates around awkwardly
- You can no longer touch the health pool

**Comment by Tomerd-afk** [comment]
> *

What about the car movement do you want changed? Once I know I can fix it and update the PR

**Comment by toryjleo** [comment]
> > 
> 
> What about the car movement do you want changed? Once I know I can fix it and update the PR

Upon further examination, I think it is good enough for the moment.

File: Assets/Resources/AiClasses/CarAi.cs [hunk 7]
```diff
@@ -0,0 +1,94 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CarAi : Ai
```
Comment (toryjleo) [inline Assets/Resources/AiClasses/CarAi.cs:5]: Please leave class comment

File: Assets/Scripts/AI/CarMovementComponent.cs [hunk 8]
```diff
@@ -0,0 +1,53 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CarMovementComponent : MovementComponent
```
Comment (toryjleo) [inline Assets/Scripts/AI/CarMovementComponent.cs:5]: Please leave class comment

**Review by toryjleo — APPROVED** [review]
LGTM :shipit:

File: Assets/Scripts/AI/Ai.cs [hunk 9]
```diff
@@ -102,12 +102,14 @@ public void Die()
             //Notify all listeners that this AI has died
             DeadEvent?.Invoke();
 
-            animationStateController.TriggerDeathA();//TODO: add catch
+            rb.detectCollisions = false;
 
-            animationStateController.TriggerDeathA();
+            if(animationStateController != null)
+            {
+                animationStateController.TriggerDeathA();//TODO: add catch
```
Comment (LiamMillerSE) [inline Assets/Scripts/AI/Ai.cs:109]: Is this TODO for a future commit?

**Review by LiamMillerSE — CHANGES_REQUESTED** [review]
Add the catch if needed, or merge without if not

**Comment by Tomerd-afk** [comment]
> Add the catch if needed, or merge without if not

merge without, the catch works for now

**Pull request closed without merging**